imxrt-hal/src/tempmon.rs

```diff
@@ -0,0 +1,400 @@
+#![cfg(any(feature = "imxrt1061", feature = "imxrt1062", feature = "imxrt1064"))]
```

Comment:
If we're guarding the module inclusion in `lib.rs` with features

```rust
#[cfg(any(feature = "imxrt1061", feature = "imxrt1062", feature = "imxrt1064"))]
 pub mod tempmon;
```

do the guards on this module become redundant?

---

We're not officially supporting any other i.MX RT families besides the 1060 and 1064 parts. However, a quick search through the 1010, 1015, 1020, and 1050 reference manuals indicates that this peripheral exists, and seems similar. (I also did a quick SVD / RAL comparison, and looks like all registers and fields are identical). Are these feature flags necessary? It seems that a contributor who wants to support the 1010 should be able to use this peripheral without issue.

It's OK with me to keep, since this reflects how it was developed and tested. But let us know if you're aware of any limitations; it'll help us later.

Reply:
I removed it from the module. I had a bad vscode setup so rust-analyzer was confused without it, but not it works even there.